src/components/Dropdown/Dropdown.vue

```diff
@@ -30,147 +30,168 @@
               {{ group.group }}
             </DropdownMenuLabel>
 
-            <DropdownMenuItem
-              v-for="item in group.items"
-              :key="item.label"
-              as-child
-              @select="item.onClick"
-            >
-              <slot v-if="$slots.item" name="item" v-bind="{ item, close }" />
-              <component
-                v-else-if="item.component"
-                :is="item.component"
-                :active="false"
+            <template v-for="item in group.items" :key="item.label">
+              <Switch
+                v-if="item.switch"
+                class="!px-2 !py-0 h-7"
+                :label="item.label"
+                label-classes="font-normal cursor-pointer"
+                :icon="item.icon"
+                @change="item.onClick?.($event)"
               />
```

Comment:
@safwansamsudeen 
fighting styles too much is a not a good idea, let's just create a switch component without the label and import it here. merging for now, let's clean it up later.